src/fork.rs

```diff
@@ -623,44 +626,46 @@ impl ForkDetails {
 
         let url = network.to_url();
         let parsed_url = SensitiveUrl::from_str(url)
-            .unwrap_or_else(|_| panic!("Unable to parse client URL: {}", &url));
-        let client = Client::http(parsed_url)
-            .unwrap_or_else(|_| panic!("Unable to create a client for fork: {}", &url))
-            .build();
-
-        (network, client)
+            .map_err(|_| eyre!("Unable to parse client URL: {}", &url))?;
+        let builder = Client::http(parsed_url)
+            .map_err(|_| eyre!("Unable to create a client for fork: {}", &url))?;
+        Ok((network, builder.build()))
     }
 
     /// Returns transactions that are in the same L2 miniblock as replay_tx, but were executed before it.
-    pub async fn get_earlier_transactions_in_same_block(&self, replay_tx: H256) -> Vec<L2Tx> {
-        let tx_details = self
+    pub fn get_earlier_transactions_in_same_block(
+        &self,
+        replay_tx: H256,
+    ) -> eyre::Result<Vec<L2Tx>> {
+        let opt_tx_details = self
             .fork_source
             .get_transaction_by_hash(replay_tx)
-            .unwrap()
-            .unwrap();
-        let miniblock = L2BlockNumber(tx_details.block_number.unwrap().as_u32());
+            .map_err(|err| eyre!("Cannot get transaction: {:?}", err))?;
```

Comment:
```suggestion
            .map_err(|err| eyre!("Cannot get transaction to replay by hash: {:?}", err))?;
```

Reply:
I think this overflows the line, but yes, that can be reformatted...